### PR TITLE
Ruby 3 compatibility

### PR DESCRIPTION
**Pull request by tot-ioki**
This PR updates the dependencies in the gemspec and makes some minor adjustments to the code and specs to allow utilizing tickeos_b2b in newer Ruby/Rails versions (esp. ruby >= 3).

### CODE REVIEW

**Comment by tom-ioki** [comment]
I will do some testing and get back to you as soon as possible 👍

**Review by dal-ioki — APPROVED** [review]
LGTM, but I would wait for Tobi's review before merging :)

File: Gemfile.lock [hunk 1]
```diff
@@ -9,43 +9,47 @@ PATH
...
     coderay (1.1.3)
-    crack (0.4.3)
-      safe_yaml (~> 1.0.0)
-    diff-lcs (1.3)
-    faraday (1.0.1)
-      multipart-post (>= 1.2, < 3)
+    crack (0.4.5)
+      rexml
+    diff-lcs (1.5.0)
+    faraday (2.7.4)
```
Comment (tom-ioki) [inline Gemfile.lock:19]: This breaks the `basic_auth` in the client class. See this as reference: https://github.com/lostisland/faraday/issues/1317 and here's the solution (taken from the faraday documentation)

```ruby
Faraday.new(...) do |conn|
  conn.request :authorization, :basic, 'username', 'password'
end
```

**Review by tom-ioki — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (tom-ioki) [reply Gemfile.lock:19]: Fixed it 👍

**Pull request merged**